### PR TITLE
Generic edition modal

### PR DESCRIPTION
**Pull request by ptbrowne**
Since we will need to use the same kind of edition mechanism from
alerts to other notifications, it was necessary to make the edition
modal more generic.

- Extracted components to edit AccountGroup / Category / MaxThreshold
- Implement CategoryAlertEditModal through a generic EditionModal configured
  through field specifications

### CODE REVIEW

File: src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx [hunk 1]
```diff
@@ -137,6 +137,82 @@ const fieldSpecs = {
...
+        />
+        <div>{choosing ? <ChoosingSwitch choosing={choosing} /> : null}</div>
+      </Stepper>
+      <ModalFooter>
+        <ModalButtons>
+          <Button
+            theme={'secondary'}
+            onClick={onDismiss}
+            label={cancelButtonLabel(props, doc)}
+          />{' '}
```
Comment (drazik) [inline src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx]: This space should be replaced by a CSS margin

File: src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx [hunk 2]
```diff
@@ -146,86 +222,34 @@ const fieldSpecs = {
...
-
-    const handleConfirmEdit = () => {
-      onEditAlert(alert)
-    }
-
     const modalTitle = t('Settings.budget-category-alerts.edit.modal-title')
+    const okButtonLabel = (props, doc) =>
+      doc.id !== undefined
+        ? t('Settings.budget-category-alerts.edit.update-ok')
+        : t('Settings.budget-category-alerts.edit.create-ok')
```
Comment (drazik) [inline src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx]: `props` doesn't seem to be used in this function

File: src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx [hunk 3]
```diff
@@ -137,6 +137,82 @@ const fieldSpecs = {
...
+
+  const handleConfirmEdit = () => {
+    onEdit(doc)
+  }
+
+  return (
+    <Modal title={modalTitle} mobileFullscreen={true} dismissAction={onDismiss}>
+      <Stepper
+        showPercentage={false}
+        currentIndex={choosing ? 1 : 0}
```
Comment (drazik) [inline src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx]: Nit: Even if they look self explanatory at first sight, these 1 and 0 look a lot like magic values. I'm afraid we will not be comfortable with it in the future. Maybe we should put it in variables with clear names or add a comment?

File: src/components/EditionModal/EditionModal.jsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        />
+        <div>{choosing ? <ChoosingSwitch choosing={choosing} /> : null}</div>
+      </Stepper>
+      <ModalFooter>
+        <ModalButtons>
+          <Button
+            theme={'secondary'}
+            onClick={onDismiss}
+            label={cancelButtonLabel(props, doc)}
+          />{' '}
```
Comment (drazik) [inline src/components/EditionModal/EditionModal.jsx]: This space should be replaced by a CSS margin

File: src/components/EditionModal/EditionModal.jsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+  const handleConfirmEdit = () => {
+    onEdit(doc)
+  }
+
+  return (
+    <Modal title={modalTitle} mobileFullscreen={true} dismissAction={onDismiss}>
+      <Stepper
+        showPercentage={false}
+        currentIndex={choosing ? 1 : 0}
```
Comment (drazik) [inline src/components/EditionModal/EditionModal.jsx]: Nit: Even if they look self explanatory at first sight, these 1 and 0 look a lot like magic values. I'm afraid we will not be comfortable with it in the future. Maybe we should put it in variables with clear names or add a comment?

File: src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx (hunk 2 above)
Comment (ptbrowne) [reply src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx]: This was the API of EditionModal, I changed it to only use doc. We will add props later if needeed.

File: src/components/EditionModal/EditionModal.jsx (hunk 4 above)
Comment (ptbrowne) [reply src/components/EditionModal/EditionModal.jsx]: Removed, it was not used actually :)

File: src/components/EditionModal/EditionModal.jsx (hunk 5 above)
Comment (ptbrowne) [reply src/components/EditionModal/EditionModal.jsx]: Good points, [solved](https://github.com/cozy/cozy-banks/pull/1706/commits/d6d7d6828e8ec4d331fac39018cc4be68c10b74d)

**Pull request merged**